pydicom/datadict.py

```diff
@@ -520,36 +544,39 @@ def get_private_entry(tag, private_creator):
     """
     if not isinstance(tag, BaseTag):
         tag = Tag(tag)
+
     try:
         private_dict = private_dictionaries[private_creator]
-    except KeyError:
-        msg = "Private creator {0} ".format(private_creator)
-        msg += "not in private dictionary"
-        raise KeyError(msg)
+    except KeyError as exc:
+        raise KeyError(
+            f"Private creator '{private_creator}' not in the private "
+            "dictionary"
+        ) from exc
 
     # private elements are usually agnostic for
     # "block" (see PS3.5-2008 7.8.1 p44)
     # Some elements in _private_dict are explicit;
     # most have "xx" for high-byte of element
-    # Try exact key first, but then try with "xx" in block position
-    try:
-        dict_entry = private_dict[tag]
-    except KeyError:
```

Comment:
I don't think this `try` could ever work, since the private dictionary is Dict[str, Dict[...]] and at this point *tag* is a `BaseTag` (i.e. int)